pkg/machine/apple/vfkit-rest.go

```diff
@@ -0,0 +1,56 @@
+package apple
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"syscall"
+)
+
+// This code is adapted from github.com/crc-org/vfkit/pkg/rest/rest.go as of vkit v0.6.0.
+
+// see `man unix`:
+// UNIX-domain addresses are variable-length filesystem pathnames of at most 104 characters.
+func maxSocketPathLen() int {
+	var sockaddr syscall.RawSockaddrUnix
+	// sockaddr.Path must end with '\0', it's not relevant for go strings
+	return len(sockaddr.Path) - 1
+}
+
+// This is intended to be equivalent to github.com/crc-org/vfkit/pkg/rest.NewEndpoint(input).ToCmdLine()
+func restNewEndpointToCmdLine(input string) ([]string, error) {
+	uri, err := url.ParseRequestURI(input)
+	if err != nil {
+		return nil, err
+	}
+
+	switch strings.ToUpper(uri.Scheme) {
+	case "NONE":
+		return []string{}, nil
+	case "UNIX":
+		if len(uri.Path) < 1 {
+			return nil, errors.New("invalid unix uri: missing path")
+		}
+		if len(uri.Host) > 0 {
+			return nil, errors.New("invalid unix uri: host is forbidden")
+		}
+		if len(uri.Path) > maxSocketPathLen() {
+			return nil, fmt.Errorf("invalid unix uri: socket path length exceeds macOS limits")
+		}
+		return []string{"--restful-uri", fmt.Sprintf("unix://%s", uri.Path)}, nil
+	case "TCP", "HTTP":
```

Comment:
AFAIK podman always seem to use an http API socket so this could be simplified even further.

(I might need to check why we use a tcp socket at all, unix seems much better suited for local IPC)

Reply:
Yes, I see things like `mc.AppleHypervisor.Vfkit.Endpoint = localhostURI + ":" + strconv.Itoa(randPort)`, but that’s a few layers away rom the call site.

I’m all in favor of simplifying further, but I’ll leave that to Podman experts.